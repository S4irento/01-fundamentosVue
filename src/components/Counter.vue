<template>
  <h3>{{ defaultTitle }}</h3>
  <p>{{ counter }}</p>
  <div>
    <button @click="increase">+1</button>
    <button @click="decrease">-1</button>
  </div>
</template>

<script>
  export default {
    props: {
      "title": String,
      "start": {
        type: Number,
        default: 0,
        validator(value){
          return value <= 100;
        }
      }
    },
    data(){
      return {
        counter: this.start
      }
    },
    methods: {
      /** @method: incremente el valor en 1 */
      increase(){
        this.counter++;
      },
      /** @method: decrementa el valor en 1 solamente cuando el valor sea mayor a 0*/
      decrease(){
        if(this.counter > 0){
          this.counter--;
        }
      }
    },
    computed: {
      defaultTitle(){
        return this.title || 'Counter';
      }
    }

  }
</script>

<style>
  button{
    background-color: rgb(14, 206, 78);
    border: 1px solid white;
    border-radius: 5px;
    color: rgb(18, 32, 51);
    cursor: pointer;
    margin: 0 5px;
    padding: 5px 15px;
    transition: 0.3s ease-in-out;
  }

  button:hover{
    background-color: rgb(9, 139, 52);
  }
</style>