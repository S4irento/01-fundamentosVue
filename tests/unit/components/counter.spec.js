import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter';

describe('Counter Component', () => {
  test('Debe hacer match con la snapshot', () =>{
    const wrapper = shallowMount(Counter);

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Debe recibir Counter en la etiqueta p', () =>{
    const wrapper = shallowMount(Counter);

    const p = wrapper.find('h3');

    expect(p.text()).toBe('Counter');
  })
})