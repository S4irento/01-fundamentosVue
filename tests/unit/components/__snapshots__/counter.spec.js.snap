// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe hacer match con la snapshot 1`] = `
<h3>Counter</h3>
<p>0</p>
<div><button>+1</button><button>-1</button></div>
`;
